### ISSUE TITLE
TERMINAL:CHARHEIGHT changes don't affect CHARWIDTH until a physics tick later.

### ISSUE THREAD

**Issue by Dunbaratu** [KSP-KOS/KOS#2066]
Let's say your font aspect ratio is such:

```
When charheight is:    charwidth is:
          20                12
          16                10
```
If you do this:
```
set terminal:charheight to 20.
wait 0.
print "height x width is: " + terminal:charheight + " x " + terminal:charwidth.

print "okay, now changing it:".

set terminal:charheight to 16.
wait 0. // wait here.
print "height x width is: " + terminal:charheight + " x " + terminal:charwidth.
// commented-out: wait 0.
print "again!".
print "height x width is: " + terminal:charheight + " x " + terminal:charwidth.
```
You get the expected result:
```
height x width is: 20 x 12
okay now changing it:
height x width is: 16 x 10
again!
height x width is: 16 x 10
```
But if you move the second wait 0 further down, like so:
```
set terminal:charheight to 20.
wait 0.
print "height x width is: " + terminal:charheight + " x " + terminal:charwidth.

print "okay, now changing it:".

set terminal:charheight to 16.
// commented-out: wait 0.
print "height x width is: " + terminal:charheight + " x " + terminal:charwidth.
wait 0.
print "again!".
print "height x width is: " + terminal:charheight + " x " + terminal:charwidth.
```
Then you get this instead:
```
height x width is: 20 x 12
okay now changing it:
height x width is: 16 x 12
again!
height x width is: 16 x 10
```
Note, the 16 x 12 output line this time, where before it was 16 x 10.

Something is causing char width calculations to not be right until the next physics tick *after* the height changed, even though the height value changes right away.

This may seem minor, but it does cause boot scripts that try to resize the terminal get it wrong.

If they do this:
```
set terminal:charheight to 16.
set terminal:height to 20.
set terminal:width to 30.
```
Without a ``wait 0.`` after the change to charheight, then what happens is the terminal resizes itself wrongly because it is using the stale calculation of the char width when deciding the pixels size of the window.  (i.e. it resizes the window on *this* tick, using the old width, and it comes out all wrong).

**Comment by Dunbaratu** [comment]
EDIT: It seems it might not even be physics ticks it has to wait for for the value to be correct, but rather **GUI Ticks**, because it looks like the charwidth doesn't get set depending on the font and the height until the Unity IMGUI system calls ``TermWindow.TerminalGUI()``.  That may or may not be one physics tick.

**Comment by Dunbaratu** [comment]
After a very long wasted exploration of the issue, I decided to merely document it and not fix it.

It's complex why, but at the heart of it is the fact that our code base has a wall of separation between kOS.Safe (cannot make calls to Unity) and kOS (can make calls to Unity), and the places that need to get the height-to-width ratio info are all in the can't-call-Unity part of the mod, and you need Unity's font engine to find the width that goes with a height.  In the past we had a lot of dependency inversion work to make this sort of problem solvable, but in this case because the relevant methods I'd need to make accessable are static, not dynamic, C# doesn't let me work with them in this way (it can't override static members, so it can't put them inside interface definitions).

In a nutshell, it's fixable, but requires a refactor that uglifies the code more than it's worth it.  Not when it could be "fixed" by having the documentation mention explicitly that you should `wait` after changing font sizes before setting terminal char/width count or the width will be wrong.

**Comment by Dunbaratu** [comment]
It just occurred to me I *could* fix this if we're willing to let the code take an extra physics tick every time you set CHARHEIGHT.

We could enforce a YieldProgram logic here.  Once CharHieght changes, the program won't continue until after kOS has had the chance to update the CharWidth in the next GUI pass.

**Comment by Dunbaratu** [comment]
No yieldprogram necessary.  I found the correct order in which to do things to make it work properly.
